src/Control/Effect/NonDet.hs

```diff
@@ -45,99 +28,16 @@ runNonDet (NonDetC m) = m (liftA2 (<|>)) (pure . pure) (pure empty)
 --     pure (a, b, c)
 -- @
 oneOf :: (Foldable t, Alternative m) => t a -> m a
-oneOf = getAlt . foldMap (Alt . pure)
-
--- | A carrier for 'NonDet' effects based on Ralf Hinze’s design described in [Deriving Backtracking Monad Transformers](https://www.cs.ox.ac.uk/ralf.hinze/publications/#P12).
-newtype NonDetC m a = NonDetC
-  { -- | A higher-order function receiving three continuations, respectively implementing '<|>', 'pure', and 'empty'.
-    runNonDetC :: forall b . (m b -> m b -> m b) -> (a -> m b) -> m b -> m b
-  }
-  deriving (Functor)
-
--- $
---   prop> run (runNonDet (pure a *> pure b)) === Just b
---   prop> run (runNonDet (pure a <* pure b)) === Just a
-instance Applicative (NonDetC m) where
-  pure a = NonDetC (\ _ leaf _ -> leaf a)
-  {-# INLINE pure #-}
-  NonDetC f <*> NonDetC a = NonDetC $ \ fork leaf nil ->
-    f fork (\ f' -> a fork (leaf . f') nil) nil
-  {-# INLINE (<*>) #-}
-
--- $
---   prop> run (runNonDet (pure a <|> (pure b <|> pure c))) === Fork (Leaf a) (Fork (Leaf b) (Leaf c))
---   prop> run (runNonDet ((pure a <|> pure b) <|> pure c)) === Fork (Fork (Leaf a) (Leaf b)) (Leaf c)
-instance Alternative (NonDetC m) where
-  empty = NonDetC (\ _ _ nil -> nil)
-  {-# INLINE empty #-}
-  NonDetC l <|> NonDetC r = NonDetC $ \ fork leaf nil -> fork (l fork leaf nil) (r fork leaf nil)
-  {-# INLINE (<|>) #-}
-
-instance Monad (NonDetC m) where
-  NonDetC a >>= f = NonDetC $ \ fork leaf nil ->
-    a fork (\ a' -> runNonDetC (f a') fork leaf nil) nil
-  {-# INLINE (>>=) #-}
-
-instance Fail.MonadFail m => Fail.MonadFail (NonDetC m) where
-  fail s = lift (Fail.fail s)
-  {-# INLINE fail #-}
-
-instance MonadFix m => MonadFix (NonDetC m) where
-  mfix f = NonDetC $ \ fork leaf nil ->
-    mfix (\ a -> runNonDetC (f (fromJust (fold (<|>) Just Nothing a)))
-      (liftA2 Fork)
-      (pure . Leaf)
-      (pure Nil))
-    >>= fold fork leaf nil
-  {-# INLINE mfix #-}
-
-instance MonadIO m => MonadIO (NonDetC m) where
-  liftIO io = lift (liftIO io)
-  {-# INLINE liftIO #-}
+oneOf = foldMapA pure
 
-instance MonadPlus (NonDetC m)
+-- | Map a 'Foldable' collection of values into a nondeterministic computation using the supplied action.
+foldMapA :: (Foldable t, Alternative m) => (a -> m b) -> t a -> m b
+foldMapA f = getAlt #. foldMap (Alt #. f)
```

Comment:
Snuck this in too. I end up defining this in just about every project that I use `Alternative` in; I’m considering proposing its addition to `base` (tho prolly under a different name, maybe `asumMap` because `concat` : `concatMap` :: `asum` : `asumMap`).